lake/index/writer.go

```diff
@@ -14,18 +14,19 @@ import (
 	"github.com/brimdata/zed/pkg/iosrc"
 	"github.com/brimdata/zed/zbuf"
 	"github.com/brimdata/zed/zng"
+	"github.com/brimdata/zed/zqe"
 	"github.com/brimdata/zed/zson"
 )
 
-func NewWriter(ctx context.Context, u iosrc.URI, def *Definition) (*Writer, error) {
+func NewWriter(ctx context.Context, path iosrc.URI, ref Reference) (*Writer, error) {
```

Comment:
`Reference` is big. Use a pointer.
```suggestion
func NewWriter(ctx context.Context, path iosrc.URI, ref *Reference) (*Writer, error) {
```